packages/gatsby/src/internal-plugins/webpack-theme-component-shadowing/index.js

```diff
@@ -119,7 +119,7 @@ module.exports = class GatsbyThemeComponentShadowingResolverPlugin {
           .map(({ themeDir }) => path.join(themeDir, `src`, theme))
       )
       .map(dir => path.join(dir, component))
-      .find(possibleComponentPath => {
+      .map(possibleComponentPath => {
```

Comment:
My two cents: this was already kind of confusing and this change did not help. 
- the change from `.find` to `.map` is a red flag
- the back-to-back `.map` calls look a little redundant 
- .find(Boolean) is a new one for me

I think this whole return statement-concat-map-map-find chunk ought to be rewritten to be a regular loop.

Up to you.

Reply:
100% agree. It's on our radar to rewrite all of this, but didn't want to bundle that with the bug fix.